lib/opam_build.ml

```diff
@@ -74,28 +74,33 @@ let install_project_deps ~opam_files ~selection =
       []
   in
   let network = ["host"] in
+  let opam_cmd = "/usr/bin/opam-" ^ (Opam_version.to_string opam_version) in
+  let root_pkgs = String.concat " " root_pkgs in
+  let non_root_pkgs = String.concat " " non_root_pkgs in
   (if Variant.arch variant |> Ocaml_version.arch_is_32bit then
      [shell ["/usr/bin/linux32"; "/bin/sh"; "-c"]] else [])
   @ distro_extras @ [
+    run "sudo ln -f %s /usr/bin/opam" opam_cmd;
     workdir "/src";
     run "sudo chown opam /src";
     run ~network ~cache
       "cd ~/opam-repository && \
        (git cat-file -e %s || git fetch origin master) && \
        git reset -q --hard %s && git log --no-decorate -n1 --oneline \
-       && opam update -u" commit commit;
-  ] @ pin_opam_files ~network groups @ [
-    env "DEPS" (String.concat " " non_root_pkgs);
-    run ~network ~cache "opam depext --update -y %s $DEPS" (String.concat " " root_pkgs);
-    run ~network ~cache "opam install $DEPS"
-  ]
+       && opam update -u" commit commit ]
+  @ pin_opam_files ~network groups
+  @ [ env "DEPS" non_root_pkgs ]
+  @ match opam_version with
+    | `V2_0 -> [run ~network ~cache "opam depext --update -y %s $DEPS" root_pkgs]
+    | `V2_1 -> []
```

Comment:
How do the depexts of the root packages get installed in the new scheme?

Reply:
Automatically as part of `opam install`

Reply:
But the `opam install` below doesn't install `root_pkgs` (only `$DEPS`).